src/MassTransit/Configuration/DefaultEndpointNameFormatter.cs

```diff
@@ -21,17 +21,29 @@ public class DefaultEndpointNameFormatter :
         static readonly char[] _removeChars = { '.', '+' };
 
         static readonly Regex _nonAlpha = new Regex("[^a-zA-Z0-9]", RegexOptions.Compiled | RegexOptions.Singleline);
-        readonly bool _includeNamespace;
-        readonly string _prefix;
-        readonly string _joinSeparator;
+
+        /// <summary>
+        /// Gets a value indicating whether the namespace is included in the name.
+        /// </summary>
+        protected internal bool IncludeNamespace { get; }
```

Comment:
why those changes? you can easily inherit from this class and provide whatever you like in ctor

Reply:
I'm fine with that, but `protected internal` is not necessary, just `protected` would be more than enough.

Reply:
> why those changes? you can easily inherit from this class and provide whatever you like in ctor

The idea was to have all members available in the sub class which are used in `FormatName`. If you directly inherit from DefaultEndpointNameFormatter you might be able to store a duplicate of those fields in the subclass to have them available later. But beside this not-so-nice duplication of class members, you might inherit from a different formatter which has fixed base constructor parameters set. e.g. we have a custom (shared) endpoint formatter which sets fixed values in the base constructor call. Somebody else in the company reusing could not extend our class reliably without relying on insights how the base implementation is done. This implies a risk if we change the base implementation that extensions would break.  

> I'm fine with that, but `protected internal` is not necessary, just `protected` would be more than enough.

A mistake from my side. I mixed up the default visibility modifier when none specified. I wrongly had in mind that the default visibility for class members is internal if nothing specified  (like it is for top level types) but it is private. Will change that :)